snowfakery/data_generator_runtime.py

```diff
@@ -356,22 +346,38 @@ def __init__(
             if not name.startswith("_") and name != "context"
         }
 
-        self.templates = templates
+        self.statements = statements
+
+    def faker_template_library(self, locale):
+        rc = self.faker_template_libraries.get(locale)
+        if not rc:
+            rc = FakerTemplateLibrary(self.faker_providers, locale)
+            self.faker_template_libraries[locale] = rc
+        return rc
 
     def loop_over_templates_until_finished(self, continuing):
         finished = False
         runtimecontext = RuntimeContext(interpreter=self)
```

Comment:
It was not obvious to me that constructing a RuntimeContext updates the interpreter's `current_context`. Perhaps that could be made explicit?
```suggestion
        self.current_context = RuntimeContext(interpreter=self)
```

(And remove the implicit setting within RuntimeContext's `__init__`)